server/controllers/auth.js

```diff
@@ -0,0 +1,61 @@
+const User = require("../models/User");
+const ToDoList = require("../models/ToDoList");
+const ToDo = require("../models/ToDo");
+const {StatusCodes} = require("http-status-codes");
+const {BadRequestError, UnauthenticatedError} = require("../errors");
+
+const register = async (req, res) => {
+  const user = await User.create({...req.body});
+  const token = user.createJWT();
+  res
+    .status(StatusCodes.CREATED)
+    .json({user: {username: user.username}, token});
+};
+
+const login = async (req, res) => {
+  const {username, password} = req.body;
+
+  if (!username || !password) {
+    throw new BadRequestError("Please provide username and password.");
+  }
+  const user = await User.findOne({username});
+  if (!user) {
+    throw new UnauthenticatedError("Invalid credentials. No such user.");
+  }
+  const isPasswordCorrect = await user.comparePassword(password);
+  if (!isPasswordCorrect) {
+    throw new UnauthenticatedError("Invalid credentials. Incorrect password.");
+  }
+
+  const token = user.createJWT();
+  res.status(StatusCodes.OK).json({user: {username: user.username}, token});
+};
+
+const deactivate = async (req, res) => {
```

Comment:
Help me out here, Danielle. Facebook call removing one's account "Delete account," Linkedin "Close account," and Twitter "Deactivate account." Which one you think would help you, or any other team members, understand what this controller function does?
![1294A9F7-53B6-4EF2-A7D5-8D770675891E 2](https://user-images.githubusercontent.com/58524911/198850771-06a238c0-a623-4e2b-b636-46aa84c6f873.jpg)

Reply:
I view deactivate and delete as two different states. Deactivate, to me, would mean my account is closed as in no longer accessible to myself of others, but would be possible to "reactivate" and pick up where I left off, none of my data would be deleted. 
Delete, to me, means all data associated is trashed, no longer available, unable to be retrieved.

It looks to me like this method is used to delete to do items or entire to do lists (on line 49) - meaning they will not be stored any longer. I would rename the method to "delete" instead of "deactivate" because it appears that you are actually deleting the data and it is no longer accessible at all.

Reply:
This `deactivate` is not a method; it's a controller ___function___. To tell a function from a method in JavaScript, just see if you invoke it with an object and a "." (as in `nameString.toUpperCase()`). It's a method if you do. You invoke a function as is; in this case, `deactivate(req, res)`.

As you seem insistent, the function is renamed `closeAccount` in the coming `server-sprint-2` branch, for although the other `register` and `login` functions in `/server/controllers/auth.js`, by their names, all convey the notion of action on a user account, both `close` and 'delete' are generic words that do not.

Please continue to be opinionated to facilitate discussions. Please also have in mind, I'd suggest, in raising any issue on review with the naming of a code module that does not have any functional deficiencies, that you hold the privilege of exercising approval and that in this instance the privilege might have been exercised solely based on your opinion that might or might not be agreed to in general.

Reply:
I proposed the change to the name of this function in order to make our code more readable and easily understood by others. 

This function appears to delete a to do list or to do item(s). I suggest calling it delete or deleteToDos. See lines 49-53.

Renaming it closeAccount does not at all indicate the purpose or action of this function.

Reply:
Please see my comment on the PR. I missed line 51 `user.remove()` initially. Resolving this line and will approve PR.